ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Client.hs

```diff
@@ -0,0 +1,138 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE KindSignatures      #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Ouroboros.Network.Protocol.LocalStateQuery.Client (
+      -- * Protocol type for the client
+      -- | The protocol states from the point of view of the client.
+      LocalStateQueryClient(..)
+    , ClientStIdle(..)
+    , ClientStAcquiring(..)
+    , ClientStAcquired(..)
+    , ClientStQuerying(..)
+
+      -- * Execution as a typed protocol
+    , localStateQueryClientPeer
+    ) where
+
+import           Network.TypedProtocol.Core
+
+import           Ouroboros.Network.Block (Point)
+import           Ouroboros.Network.Protocol.LocalStateQuery.Type
+
+
+newtype LocalStateQueryClient block query result m a = LocalStateQueryClient {
+      runLocalStateQueryClient :: ClientStIdle block query result m a
+    }
+
+-- | In the 'StIdle' protocol state, the client has agency and must send:
+--
+--  * a request to acquire a state
+--  * a termination messge
+--
+data ClientStIdle block query result (m :: * -> *) a where
+  SendMsgAcquire :: Point block
+                 -> ClientStAcquiring block query result m a
+                 -> ClientStIdle      block query result m a
+
+  SendMsgDone    :: a
+                 -> ClientStIdle      block query result m a
+
+-- | In the 'StAcquiring' protocol state, the client does not have agency.
+-- Instead it is waiting for:
+--
+--  * acquired
+--  * failure to acquire
+--
+-- It must be prepared to handle either.
+--
+data ClientStAcquiring block query result m a = ClientStAcquiring {
+      recvMsgAcquired ::    ClientStAcquired block query result m a,
```

Comment:
could you align the type

Reply:
Initially, I aligned all state types on their type parameters, even when some had an `m (..)` around them. Now that I have added so many `m`s, there's no point in alignment on this line, I'll remove it.